Destination/Destination.php

```diff
@@ -3,15 +3,25 @@
 namespace Zenstruck\BackupBundle\Destination;
 
 use Psr\Log\LoggerInterface;
+use \Countable;
+use \IteratorAggregate;
 
 /**
  * @author Kevin Bond <kevinbond@gmail.com>
  */
-interface Destination
+interface Destination extends Countable, IteratorAggregate
```

Comment:
Hahahahahahhaha ! Awesome ! 

You are absolutely right!

But we can have master destination, slave/mirror destinations. We list from the master, slaves gets synced. So MirroredDestination can implement same Destination interface, with additional methods:

- getMaster()
- getMirrors()
- getAll()

Underlying implementation will take care that mirrors are up to date with master.

Love the idea about mirroring! Love it! I totally forgot about that, it is really useful for paranoid sys admins :-)

Sent from my iPad

> On Sep 8, 2015, at 5:38 PM, Kevin Bond <notifications@github.com> wrote:
> 
> In Destination/Destination.php:
> 
> >  
> >  /**
> >   * @author Kevin Bond <kevinbond@gmail.com>
> >   */
> > -interface Destination
> > +interface Destination extends Countable, IteratorAggregate
> destination holds collection of backups, so it is aggregating object
> 
> Backups may not be all it holds. We could have a MirrorStreamDestination, does count($destination) count the backups or the mirrors? Meh, probably just splitting hairs but if there is no great reason to have it, let's leave it out.
> 
> —
> Reply to this email directly or view it on GitHub.
>